# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wgpu_bindings/src/server.rs
# Commit: 4bf253633e70
# Full Hash: 4bf253633e709f884f776b1c70e8c39ca30dea23
# Author: sotaro <sotaro.ikeda.g@gmail.com>
# Date: 2024-12-21 09:42:56
# Regressor Bug: 1938311
# File Overlap Count: 1
# Description:
#   Bug 1938311 - Change ExternalTexture's requested size error handling to be explicit r=webgpu-reviewers,gfx-reviewers,ErichDonGubler
#   
#   Error handling is not clear in the current code. Change to explicit error handling.
#   
#   In the current code, if the texture size is larger than max_texture_dimension_2d, self.device_create_texture() is always expected to return error.
# ==============================================================================

diff -r 84f81ea88258 -r 4bf253633e70 gfx/wgpu_bindings/src/server.rs
--- a/gfx/wgpu_bindings/src/server.rs	Sat Dec 21 02:38:32 2024 +0200
+++ b/gfx/wgpu_bindings/src/server.rs	Sat Dec 21 00:22:27 2024 +0000
@@ -1893,20 +1893,24 @@
                     return;
                 }
 
-                let mut use_external_texture = if let Some(id) = swap_chain_id {
+                let use_external_texture = if let Some(id) = swap_chain_id {
                     unsafe { wgpu_server_use_external_texture_for_swap_chain(self.owner, id) }
                 } else {
                     false
                 };
 
-                let limits = self.device_limits(self_id);
-                if desc.size.width > limits.max_texture_dimension_2d
-                    || desc.size.height > limits.max_texture_dimension_2d
-                {
-                    use_external_texture = false;
-                }
+                if use_external_texture {
+                    let limits = self.device_limits(self_id);
+                    if desc.size.width > limits.max_texture_dimension_2d
+                        || desc.size.height > limits.max_texture_dimension_2d
+                    {
+                        error_buf.init(ErrMsg {
+                            message: "size exceeds limits.max_texture_dimension_2d",
+                            r#type: ErrorBufferType::Validation,
+                        });
+                        return;
+                    }
 
-                if use_external_texture {
                     #[cfg(target_os = "windows")]
                     {
                         let is_created = self.create_texture_with_external_texture_d3d11(
